content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,669 @@
+---
+title: "Peace of Mind with Cloud Secret Providers"
+
+date: 2020-05-07T14:30:35-07:00
+draft: false
+meta_desc: "Encrypt your Pulumi secrets with a cloud secrets provider"
+meta_image: meta.png
+authors:
+    - lee-briggs
+tags:
+    - features
+    - security
+---
+
+The secrets in your infrastructure are a vital part of your security model, and provisioning infrastructure is an inherently privileged process. [Previously](https://www.pulumi.com/blog/managing-secrets-with-pulumi/) we introduced client-side secret encryption and started encrypting secret configuration values inside the Pulumi state so that users could be confident their passwords, tokens and other secret values were viewable only by them while managing their infrastructure.
+Our first iteration of the client-side encryption used a passphrase for encrypting the secret, however, remembering that passphrase and sharing it securely with team members is less than ideal.
+That's why we also added support for "Cloud Secret Providers", giving users full confidence that their sensitive values are for their eyes only.
+
+<!--more-->
+
+Pulumi supports encryption via [AWS KMS](https://aws.amazon.com/kms/), [Azure KeyVault](https://azure.microsoft.com/en-us/services/key-vault/), [Google Cloud KMS](https://cloud.google.com/kms) and [Hashicorp Vault](https://www.vaultproject.io/). This post shows you _one_ example of using a cloud secret provider in a Pulumi stack using AWS KMS.
```

Comment:
I do think it's worth noting that Pulumi supports encryption via the Pulumi Service itself - so users can get encryption of sensitive configuration and state automatically with no extra effort.  But that users who want to use their own encryption can do that with these cloud secret providers.